### PR TITLE
Bug 1800700 - Add default search suggest header

### PR DESCRIPTION
**Pull request by sarah541**
### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### QA

- [x] **QA Needed**

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-debug` task.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation

Used by GitHub Actions.

### CODE REVIEW

**Comment by sarah541** [comment]
Screen Shot 2022-11-17 at 1 01 12 AM

**Comment by sarah541** [comment]
To be merged after landing this AC pr : https://github.com/mozilla-mobile/firefox-android/pull/157

File: app/src/main/java/org/mozilla/fenix/search/awesomebar/AwesomeBarView.kt [hunk 1]
```diff
@@ -207,6 +212,18 @@ class AwesomeBarView(
         searchSuggestionProviderMap = HashMap()
     }
 
+    private fun getDefaultSearchEngine(components: Components): String? {
+        var defaultSearchEngine = components.core.store.state.search.selectedOrDefaultSearchEngine?.name
+
+        if (!defaultSearchEngine.isNullOrEmpty()) {
+            defaultSearchEngine = when (defaultSearchEngine) {
+                HomeFragment.GOOGLE_SEARCH_ENGINE_NAME -> "$defaultSearchEngine Search"
+                else -> "$defaultSearchEngine search"
```
Comment (gabrielluong) [inline app/src/main/java/org/mozilla/fenix/search/awesomebar/AwesomeBarView.kt:221]: I would imagine we need to localized these strings here. You can have strings that can take a parameter (eg, defaultSearchEngine). See https://searchfox.org/mozilla-mobile/source/fenix/app/src/main/res/values/strings.xml#435 and https://searchfox.org/mozilla-mobile/source/fenix/app/src/main/java/org/mozilla/fenix/settings/HomeSettingsFragment.kt#106.

When you define the string takes a parameter, you also need to specify what the parameter is in localization comment.

File: app/src/main/java/org/mozilla/fenix/search/awesomebar/AwesomeBarView.kt [hunk 2]
```diff
@@ -207,6 +212,18 @@ class AwesomeBarView(
         searchSuggestionProviderMap = HashMap()
     }
 
+    private fun getDefaultSearchEngine(components: Components): String? {
```
Comment (rocketsroger) [inline app/src/main/java/org/mozilla/fenix/search/awesomebar/AwesomeBarView.kt:215]: For me this function name is misleading.  This is not getting the default search engine but the title to use with the default search engine?

**Comment by gabrielluong** [comment]
@sarah541 Could you add a screenshot of what this looks like?

File: app/src/main/res/values/strings.xml [hunk 3]
```diff
@@ -1810,6 +1810,14 @@
     <!-- Text for the snackbar to confirm auto-close is enabled for inactive tabs -->
     <string name="inactive_tabs_auto_close_message_snackbar">Auto-close enabled</string>
 
+    <!-- Awesome bar suggestions' headers -->
+    <!-- Title for firefox suggestion -->
+    <string name="firefox_suggest_header">Firefox Suggest</string>
+    <!-- Title for suggestions when Google is default search suggestion engine. The first parameter is Google search engine. -->
+    <string name="google_search_engine_suggestion_header">%s Search</string>
```
Comment (t-p-white) [inline app/src/main/res/values/strings.xml:1817]: Nit: I wonder if the parameter is necessary for this string as it's exclusively used for Google

File: app/src/main/res/values/strings.xml [hunk 4]
```diff
@@ -1847,7 +1855,6 @@
     <string name="experiments_snackbar">Enable telemetry to send data.</string>
     <!-- Snackbar button text to navigate to telemetry settings.-->
     <string name="experiments_snackbar_button">Go to settings</string>
-    <string name="firefox_suggest_header">Firefox Suggest</string>
```
Comment (gabrielluong) [inline app/src/main/res/values/strings.xml:1850]: Spoke with Delphine and moving strings around is OK.

File: app/src/main/res/values/strings.xml (hunk 3 above)
Comment (gabrielluong) [reply app/src/main/res/values/strings.xml:1817]: @t-p-white This was a great catch. Thank you. 

You did mention on Slack that you were inspecting how other apps for this string. I was wondering how did you go about that?

**Comment by gabrielluong** [comment]
Closing this for https://github.com/mozilla-mobile/fenix/pull/27928 since Sarah's away on PTO.

File: app/src/main/res/values/strings.xml (hunk 3 above)
Comment (t-p-white) [reply app/src/main/res/values/strings.xml:1817]: @gabrielluong in AS, in th 'root' strings file if you hover over the strings name it pops up a dialog of all the translations 🙂

**Pull request closed without merging**